shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -236,28 +240,26 @@ - (void)sharedSetupWithProject:(nullable FlutterDartProject*)project
   // Need the project to get settings for the view. Initializing it here means
   // the Engine class won't initialize it later.
   if (!project) {
-    project = [[[FlutterDartProject alloc] init] autorelease];
+    project = [[FlutterDartProject alloc] init];
   }
   FlutterView.forceSoftwareRendering = project.settings.enable_software_rendering;
   _weakFactory = std::make_unique<fml::WeakNSObjectFactory<FlutterViewController>>(self);
-  auto engine = fml::scoped_nsobject<FlutterEngine>{[[FlutterEngine alloc]
-                initWithName:@"io.flutter"
-                     project:project
-      allowHeadlessExecution:self.engineAllowHeadlessExecution
-          restorationEnabled:[self restorationIdentifier] != nil]};
-
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"io.flutter"
+                                                      project:project
+                                       allowHeadlessExecution:self.engineAllowHeadlessExecution
+                                           restorationEnabled:self.restorationIdentifier != nil];
   if (!engine) {
     return;
   }
 
   _viewOpaque = YES;
   _engine = engine;
-  _flutterView.reset([[FlutterView alloc] initWithDelegate:_engine
-                                                    opaque:self.isViewOpaque
-                                           enableWideGamut:project.isWideGamutEnabled]);
-  [_engine.get() createShell:nil libraryURI:nil initialRoute:initialRoute];
+  _flutterView = [[FlutterView alloc] initWithDelegate:_engine
+                                                opaque:_viewOpaque
+                                       enableWideGamut:project.isWideGamutEnabled];
+  [_engine createShell:nil libraryURI:nil initialRoute:initialRoute];
   _engineNeedsLaunch = YES;
-  _ongoingTouches.reset([[NSMutableSet alloc] init]);
+  _ongoingTouches = [[NSMutableSet alloc] init];
```

Comment:
Even though this isn't technically an initialiser, it's initialiser-like enough that I've elected to reference the ivars directly rather than use property notation. It should be possible to migrate, but I'm being paranoid here.

Switched the once self.isViewOpaque usage to an ivar usage for consistency.

I'll likely give this another pass once the ARC migration is complete.

Reply:
> Even though this isn't technically an initialiser, it's initialiser-like enough

It's currently being called from `init*`, so it's effectively initializer code.

The real problem of course is that an initializer is calling helper instance methods in the first place. In addition to creating this kind of confusion, it's inherently dangerous because a subclass could (even accidentally) override this method.

> It should be possible to migrate, but I'm being paranoid here.
> [...]
> I'll likely give this another pass once the ARC migration is complete.

The best option for code like this is to make it a static C function (so that it can't be overridden) that takes `self` as an argument, and is clearly named and documented as doing shared initialization. It makes the code slightly uglier, but arguably even that is a feature since it means there's another warning sign that it's unusual code that needs special handling.

(For this PR, your changes LG; I would also put an implementation comment on the method warning that it's called from an initializer and so should handle ivars accordingly, and with a TODO for the above.)